exec/java-exec/src/test/java/org/apache/drill/test/LogFixture.java

```diff
@@ -0,0 +1,255 @@
+    }
+
+    /**
+     * Set a specific logger to the given level.
+     *
+     * @param loggerName name of the logger (typically used for package-level
+     * loggers)
+     * @param level the desired Logback-defined level
+     * @return this builder
+     */
+    public LogFixtureBuilder logger(String loggerName, Level level) {
+      loggers.add(new LogSpec(loggerName, level));
+      return this;
+    }
+
+    /**
+     * Set a specific logger to the given level.
+     *
+     * @param loggerClass class that defines the logger (typically used for
+     * class-specific loggers)
+     * @param level the desired Logback-defined level
+     * @return this builder
+     */
+    public LogFixtureBuilder logger(Class<?> loggerClass, Level level) {
+      loggers.add(new LogSpec(loggerClass.getName(), level));
+      return this;
+    }
+
+    /**
+     * Turns off all logging. If called first, you can set disable as your
+     * general policy, then turn back on loggers selectively for those
+     * of interest.
+     * @return this builder
+     */
+    public LogFixtureBuilder disable() {
+      return rootLogger(Level.OFF);
+    }
+
+    /**
+     * Set the desired log level on the root logger.
+     * @param level the desired Logback log level
+     * @return this builder
+     */
+
+    public LogFixtureBuilder rootLogger(Level level) {
+      loggers.add(new LogSpec(Logger.ROOT_LOGGER_NAME, level));
+      return this;
+    }
+
+    /**
+     * Apply the log levels and output, then return a fixture to be used
+     * in a try-with-resources block. The fixture automatically restores
+     * the original configuration on completion of the try block.
+     * @return the log fixture
+     */
+    public LogFixture build() {
+      return new LogFixture(this);
+    }
+  }
+
+  private PatternLayoutEncoder ple;
+  private ConsoleAppender<ILoggingEvent> appender;
+  private List<LogSpec> loggers = new ArrayList<>();
+  private Logger drillLogger;
+
+  public LogFixture(LogFixtureBuilder builder) {
+    if (builder.logToConsole) {
+      setupConsole(builder);
+    }
+    setupLoggers(builder);
+  }
+
+  /**
+   * Creates a new log fixture builder.
+   * @return the log fixture builder
+   */
+
+  public static LogFixtureBuilder builder() {
+    return new LogFixtureBuilder();
+  }
+
+  private void setupConsole(LogFixtureBuilder builder) {
+    Logger drillLogger = (Logger)LoggerFactory.getLogger(DRILL_PACKAGE_NAME);
+    if (drillLogger.getAppender("STDOUT") != null) {
+      return;
+    }
+    LoggerContext lc = (LoggerContext) LoggerFactory.getILoggerFactory();
+    ple = new PatternLayoutEncoder();
+    ple.setPattern(builder.consoleFormat);
+    ple.setContext(lc);
+    ple.start();
+
+    appender = new ConsoleAppender<>( );
+    appender.setContext(lc);
+    appender.setName("Console");
+    appender.setEncoder( ple );
+    appender.start();
+
+    Logger root = (Logger)LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
```

Comment:
We are checking only for console appender in DrillLogger. Can't root logger have the console appender and drillLogger may not ? In which case we will overwrite the console appender of root logger and while restoring we will remove the console appender from both.